### PR TITLE
Drop Carthage support 🔥

### PR DESCRIPTION
**Pull request by scannillo**
### Motivation

- Carthage has been falling behind in supporting the latest iOS dependency requirements (ex: `xcframework`)
- There are [numerous Carthage bugs](https://github.com/Carthage/Carthage/releases/tag/0.37.0) in the latest release of Carthage for frameworks with nested dependencies 
- Many of our merchants are asking for SPM support, and we are choosing to focus our efforts on that
    - With the release of CardinalMobile's `xcframework` our SPM integration story is extremely simplified

### Summary of changes

 - Remove CarthageTest sample app
 - Remove carthage tasks from Rakefile & GH Actions .yml files
 - Remove Carthage references in README 

 ### Checklist

 - [X] Added a changelog entry

### Authors
@scannillo @sestevens

### CODE REVIEW

**Review by sarahkoop — APPROVED** [review]
🌟

**Review by sshropshire — APPROVED** [review]

File: Cartfile.private [hunk 1]
```diff
@@ -1 +1,2 @@
-github "https://github.com/futuretap/InAppSettingsKit.git" ~> 3.0
+github "braintree/braintree_ios" ~> 5.3
```
Comment (sshropshire) [inline Cartfile.private:1]: random: is this `Cartfile.private` file still needed?

Comment (scannillo) [reply Cartfile.private:1]: We're still using Carthage in our demo app. We use Carthage to fetch InAppSettingsKit & Braintree dependencies in our Demo app. (The Demo app pulls in the local version of BraintreeDropIn).

We might update the Demo to use SPM in a future PR.

Comment (sshropshire) [reply Cartfile.private:1]: Ahh TIL

**Pull request merged**